packages/redux/src/entities/selectors/entity.ts

```diff
@@ -42,36 +40,27 @@ export const getEntities = <
 /**
  * Gets a specific entity of the given schema name, by id.
  *
- * @memberof module:entities/selectors
- *
- * @param {object} state - Application state.
- * @param {string} name - Entity/schema name.
- * @param {string|number} id - Entity identifier.
+ * @param state - Application state.
+ * @param name - Entity/schema name.
+ * @param id - Entity identifier.
  *
- * @returns {object|undefined} - The specific entity of the given id, undefined
+ * @returns - The specific entity of the given id, undefined
  * if it isn't found.
  */
-export const getEntityById = <
-  SchemaName extends keyof NonNullable<StateEntities>,
-  EntityId extends keyof NonNullable<NonNullable<StateEntities>[SchemaName]>,
->(
+export const getEntityById = <T>(
```

Comment:
Why change this to "<T>"? You don't seem to be using it anywhere 🤔. This PR is a little confusing, you say you're "fixing" the selector and its errors but I can't find what those problems are 😕

Reply:
The <T> means that it's using the type from the call. Which is being used in the return type. For example this call:

```javascript
export const getBrand = (
  state: StoreState,
  brandId: Brand['id'],
): Brand | undefined => getEntityById(state, 'brands', brandId);
```

It's passing the `Brand | undefined` as the response type for the getEntityById. Without this the response type was always unknown or any and every single call had an error because the type was unknown. This was the main problem with the function, the rest was just to clean it up a little bit.

Reply:
Got it, thanks!